src/runtime/getPositions.ts

```diff
@@ -153,33 +170,52 @@ function getDisplayProps(
 async function resolveAppTokenPosition(
   address: string,
   positionDefinition: AppTokenPositionDefinition & { appId: string },
-  tokensByAddress: TokensInfo,
-  resolvedTokens: Record<string, Omit<Token, 'balance'>>,
+  tokensByTokenId: TokensInfo,
+  resolvedTokensByTokenId: Record<string, Omit<Token, 'balance'>>,
   appInfo: AppInfo,
 ): Promise<AppTokenPosition> {
   let pricePerShare: DecimalNumber[]
   if (typeof positionDefinition.pricePerShare === 'function') {
     pricePerShare = await positionDefinition.pricePerShare({
-      tokensByAddress,
-    } as PricePerShareContext)
+      tokensByTokenId,
+    })
   } else {
     pricePerShare = positionDefinition.pricePerShare
   }
 
   let priceUsd = new BigNumber(0)
   for (let i = 0; i < positionDefinition.tokens.length; i++) {
     const token = positionDefinition.tokens[i]
-    const tokenInfo = tokensByAddress[token.address]!
+    const tokenInfo =
+      tokensByTokenId[
+        getTokenId({
+          networkId: token.networkId,
+          address: token.address,
+          isNative: isNative({
+            networkId: token.networkId,
+            address: token.address,
+          }),
+        })
+      ]
```

Comment:
Seeing this code written multiple times, I think I'm now convinced we should address it in this PR and make it more succint to get a token from a network/address pair.

Maybe with a helper?

Reply:
yeah it's ugly, and not safe from bugs this way since you could accidentally pass a different address or networkId in to the two functions (getTokenId and isNative) by mistake, for the same token. I updated how getTokenId is working to call isNative itself when we aren't sure either way if a token is native or not. lmk what you think.

Reply:
Thanks, this looks good now 👍